tests/unit/specification/parser_test.py

```diff
@@ -2802,3 +2814,319 @@ def test_parse_reserved_word_as_channel_name() -> None:
+            then Options
+            with Size => Length * 8
+            if Message_Type = MT_Options
+            then Value
+            if Message_Type = MT_Value
+               and Length = 1
+            then Values
+            with Size => Length * 8
+            if Message_Type = MT_Values;
+         Data : Opaque
+            then null;
+         Option_Types : Option_Types
+            then null;
+         Options : Options
+            then null;
+         Value : ValueT
+            then null;
+         Values : Values;
+      end message;
+
+   for Message use (Data => Option);
+
+   generic
+       Channel : Channel with Readable, Writable;
+   session Session with
+       Initial => Start,
+       Final => Terminated
+   is
+       Msg : Message;
+   begin
+      state Start is
+        Msg2 : Message;
+      begin
+         Channel'Read (Msg);
+         Msg2 := Msg;
+      transition
+         goto Reply
+            if Msg'Valid = True
+            and Msg.Message_Type = MT_Data
+            and Msg.Length = 1
+         goto Terminated
+      end Start;
+      state Reply is
+      begin
+         MSg := Message'(Message_Type => MT_Data, Length => 1, Data => [2]);
+         Channel'Write (Msg);
+      transition
+         goto Terminated
+      exception
+         goto Terminated
+      end Reply;
+      state Terminated is null state;
+   end Session;
+   end Test;
+"""
+    test_spec = tmp_path / "test.rflx"
+    test_rfi = tmp_path / "test.rfi"
+    test_spec.write_text(content, encoding="utf-8")
+    test_rfi.write_text(rfi_content)
+    regex = re.compile("test.rfi:.*" + match_error, re.DOTALL)
+    if raises:
+        with pytest.raises(RecordFluxError, match=regex):
+            p.parse(test_spec)
+            p.create_model()
+    else:
+        p.parse(test_spec)
+        p.create_model()
+
+
+def test_rfi_get_size() -> None:
+    filename = Path("tmp.rflx")
+    state_a = State("A", transitions=[Transition(target="B")])
+    session = Session(
+        identifier="P::S",
+        initial="A",
+        final="B",
+        states=[
+            state_a,
+            State("B"),
+        ],
+        declarations=[],
+        parameters=[],
+        types=[],
+        location=Location(start=(1, 1), source=filename),
+    )
+    session2 = Session(
+        identifier="P::S2",
+        initial="A",
+        final="B",
+        states=[
+            state_a,
+            State("B"),
+        ],
+        declarations=[],
+        parameters=[],
+        types=[],
+        location=Location(start=(1, 1), source=filename),
+    )
+    integration = Integration()
+    session_object = {"Session": {"S": {"Buffer_Size": {"Default": 1024}}}}
+    integration.add_integration_file(filename.with_suffix(".rfi"), session_object)
+    assert integration.get_size(session, None, ID("x")) == 1024
+    assert integration.get_size(session, state_a, ID("x")) == 1024
+    assert integration.get_size(session2, None, ID("x")) == 4096
```

Comment:
That is testing `specification.integration` and not `specification.parser`, thus it is in the wrong test suite.

Reply:
I created a new file `integration_test.py` in 3c684cf25a88b85b6cf8664ddc5096cd6e4bc2a0